sound/soc/sof/sof-audio.c

```diff
@@ -638,6 +639,14 @@ int sof_set_up_pipelines(struct snd_sof_dev *sdev, bool verify)
 			if (!dai || !dai->dai_config)
 				continue;
 
+			/*
+			 * For older firmware, skip scheduler widgets in this loop,
+			 * sof_widget_setup() will be called in the 'complete pipeline' loop
+			 */
+			if (v->abi_version < SOF_ABI_VER(3, 19, 0) &&
+			    swidget->id == snd_soc_dapm_scheduler)
```

Comment:
@plbossart Im thinking we can skip the ABI check here. The order doesnt matter for newer firmware anyway. So we can set up the scheduler widget in the loop below.

Reply:
Yes it's not really necessary but if someone wants to revisit this code later they will have a non-ambiguous description of the limitation, without needing to go to the commit message. It's also useful since you have a mention in the teardown() function of a similar limitation.

Reply:
Ok